### PR TITLE
Fix skip

### PR DESCRIPTION
**Pull request by w4nderlust**
Introduces the new parameters:

- `--skip_save_model`
- `--skip_save_progress`
- `--skip_save_log`

in both experiment, predict and API

### CODE REVIEW

File: docs/api/index.html [hunk 1]
```diff
@@ -908,20 +918,37 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
...
    resume training of</li>
-<li><strong>skip_save_progress_weights</strong> (bool, default: <code>False</code>): doesn't save
-   weights after each epoch. By default Ludwig saves weights after
-   each epoch for enabling resuming of training, but if the model is
-   really big that can be time consuming and will save twice as much
-   space, use this parameter to skip it.</li>
-<li><strong>dataset_type</strong> (string, default: <code>'default'</code>): determines the type
-   of preprocessing will be applied to the data. Only <code>generic</code> is
-   available at the moment</li>
+    <li><strong>skip_save_model</strong> (bool, default: <code>False</code>): does not
```
Comment (ydudin3) [inline docs/api/index.html]: disables saving model weights ...

File: docs/api/index.html [hunk 2]
```diff
@@ -908,20 +918,37 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
...
+        save model weights and hyperparameters each time the model
+        improves. By default Ludwig saves model weights after each epoch
+        the validation measure imrpvoes, but if the model is really big
+        that can be time consuming if you do not want to keep
+        the weights and just find out what performance can a model get
+        with a set of hyperparameters, use this parameter to skip it,
+        but the model will not be loadable later on.
+    </li>
+    <li><strong>skip_save_progress</strong> (bool, default: <code>False</code>): doesn't save
```
Comment (ydudin3) [inline docs/api/index.html]: disables saving progress ...

File: docs/api/index.html [hunk 3]
```diff
@@ -908,20 +918,37 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
...
+        progress each epoch. By default Ludwig saves weights and stats
+        after each epoch for enabling resuming of training, but if
+        the model is really big that can be time consuming and will uses
+        twice as much space, use this parameter to skip it, but training
+        cannot be resumed later on.
+    </li>
+    <li><strong>skip_save_log</strong> (bool, default: <code>False</code>): does not save TensorBoard
```
Comment (ydudin3) [inline docs/api/index.html]: disables saving ...

File: docs/user_guide/index.html [hunk 4]
```diff
@@ -2379,12 +2379,25 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
                         path of a pretrained model to load as initialization
   -mrp MODEL_RESUME_PATH, --model_resume_path MODEL_RESUME_PATH
                         path of a the model directory to resume training of
-  -sspw SKIP_SAVE_PROGRESS_WEIGHTS, --skip_save_progress_weights SKIP_SAVE_PROGRESS_WEIGHTS
-                        doesn&#39;t save weights after each epoch. By default
+  -ssm, --skip_save_model
+                        does not save weights each time the model imrpoves. By
```
Comment (ydudin3) [inline docs/user_guide/index.html]: disables saving ...

File: docs/user_guide/index.html [hunk 5]
```diff
@@ -2379,12 +2379,25 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
...
+                        default Ludwig saves weights after each epoch the
+                        validation measure imrpvoes, but if the model is
+                        really big that can be time consuming if you do not
+                        want to keep the weights and just find out what
+                        performance can a model get with a set of
+                        hyperparameters, use this parameter to skip it.
+  -ssp, --skip_save_progress
+                        does not save weights after each epoch. By default
```
Comment (ydudin3) [inline docs/user_guide/index.html]: disables saving ...

File: docs/user_guide/index.html [hunk 6]
```diff
@@ -2379,12 +2379,25 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
...
                         ludwig saves weights after each epoch for enabling
                         resuming of training, but if the model is really big
                         that can be time consuming and will save twice as much
                         space, use this parameter to skip it.
+  -ssl, --skip_save_log
+                        does not save TensorBoard logs. By default Ludwig
```
Comment (ydudin3) [inline docs/user_guide/index.html]: disables saving ...

File: docs/user_guide/index.html [hunk 7]
```diff
@@ -2394,6 +2407,7 @@ <h3 id="train">train<a class="headerlink" href="#train" title="Permanent link">&
                         list of gpus to use
   -gf GPU_FRACTION, --gpu_fraction GPU_FRACTION
                         fraction of gpu memory to initialize the process with
+  -uh, --use_horovod    uses horovod for distributed training
```
Comment (ydudin3) [inline docs/user_guide/index.html]: use horovod ...

File: ludwig/experiment.py [hunk 8]
```diff
@@ -124,10 +126,27 @@ def experiment(
            far are also resumed effectively cotinuing a previously interrupted
            training process.
     :type model_resume_path: filepath (str)
-    :param skip_save_progress_weights: Skips saving the weights at the end of
-           each epoch. If this is true, training cannot be resumed from the
-           exactly the state at the end of the previous epoch.
-    :type skip_save_progress_weights: Boolean
+    :param skip_save_model: Does not
```
Comment (ydudin3) [inline ludwig/experiment.py]: Disables saving ...

File: ludwig/experiment.py [hunk 9]
```diff
@@ -124,10 +126,27 @@ def experiment(
...
+           save model weights and hyperparameters each time the model
+           improves. By default Ludwig saves model weights after each epoch
+           the validation measure imrpvoes, but if the model is really big
+           that can be time consuming if you do not want to keep
+           the weights and just find out what performance can a model get
+           with a set of hyperparameters, use this parameter to skip it,
+           but the model will not be loadable later on.
+    :type skip_save_model: Boolean
+    :param skip_save_progress: Doesn't save
```
Comment (ydudin3) [inline ludwig/experiment.py]: Disables saving ...

File: ludwig/experiment.py [hunk 10]
```diff
@@ -124,10 +126,27 @@ def experiment(
...
+           progress each epoch. By default Ludwig saves weights and stats
+           after each epoch for enabling resuming of training, but if
+           the model is really big that can be time consuming and will uses
+           twice as much space, use this parameter to skip it, but training
+           cannot be resumed later on.
+    :type skip_save_progress: Boolean
+    :param skip_save_log: Does not save TensorBoard
```
Comment (ydudin3) [inline ludwig/experiment.py]: Disables saving ...

File: ludwig/experiment.py [hunk 11]
```diff
@@ -485,12 +507,39 @@ def cli(sys_argv):
...
-        '--skip_save_progress_weights',
-        help='does not save weights after each epoch. By default Ludwig saves '
-             'weights after each epoch for enabling resuming of training, but '
-             'if the model is really big that can be time consuming and will '
-             'use twice as much storage space, use this parameter to skip it.'
+        '-ssm',
+        '--skip_save_model',
+        action='store_true',
+        default=False,
+        help='does not save model weights and hyperparameters each time '
```
Comment (ydudin3) [inline ludwig/experiment.py]: disables saving ...

File: ludwig/experiment.py [hunk 12]
```diff
@@ -485,12 +507,39 @@ def cli(sys_argv):
...
+             'the weights and just find out what performance can a model get '
+             'with a set of hyperparameters, use this parameter to skip it,'
+             'but the model will not be loadable later on.'
+    )
+    parser.add_argument(
+        '-ssp',
+        '--skip_save_progress',
+        action='store_true',
+        default=False,
+        help='does not save progress each epoch. By default Ludwig saves '
```
Comment (ydudin3) [inline ludwig/experiment.py]: disables saving ...

File: ludwig/experiment.py [hunk 13]
```diff
@@ -485,12 +507,39 @@ def cli(sys_argv):
...
+             'time consuming and will uses twice as much space, use '
+             'this parameter to skip it, but training cannot be resumed '
+             'later on. '
+    )
+    parser.add_argument(
+        '-ssl',
+        '--skip_save_log',
+        action='store_true',
+        default=False,
+        help='does not save TensorBoard logs. By default Ludwig saves '
```
Comment (ydudin3) [inline ludwig/experiment.py]: disables saving ...

File: ludwig/models/model.py [hunk 14]
```diff
@@ -346,8 +349,22 @@ def train(
         :type learning_rate_warmup_epochs: Integer
         :param resume: Resume training a model that was being trained.
         :type resume: Boolean
-        :param skip_save_progress_weights:
-        :type skip_save_progress_weights:
+        :param skip_save_model: does not
```
Comment (ydudin3) [inline ludwig/models/model.py]: disables saving ...

File: ludwig/models/model.py [hunk 15]
```diff
@@ -346,8 +349,22 @@ def train(
...
+               save model weights and hyperparameters each time the model
+               improves. By default Ludwig saves model weights after each epoch
+               the validation measure imrpvoes, but if the model is really big
+               that can be time consuming if you do not want to keep
+               the weights and just find out what performance can a model get
+               with a set of hyperparameters, use this parameter to skip it,
+               but the model will not be loadable later on.
+        :type skip_save_model: Boolean
+        :param skip_save_progress:
```
Comment (ydudin3) [inline ludwig/models/model.py]: Missing annotation.

File: ludwig/models/model.py [hunk 16]
```diff
@@ -346,8 +349,22 @@ def train(
...
+        :type skip_save_progress:
+        :param skip_save_log: Does not save TensorBoard
```
Comment (ydudin3) [inline ludwig/models/model.py]: Disables saving ...

File: ludwig/models/model.py [hunk 17]
```diff
@@ -589,30 +614,36 @@ def train(
...
+                        os.path.join(
+                            save_path,
+                            TRAINING_PROGRESS_FILE_NAME
+                        ),
+                        progress_tracker
+                    )
+                    if skip_save_model:
+                        self.save_hyperparameters(
+                            self.hyperparameters,
+                            model_hyperparameters_path)
```
Comment (ydudin3) [inline ludwig/models/model.py]: Move closing parenthesis to the line below.

File: ludwig/models/model.py [hunk 18]
```diff
@@ -936,8 +967,8 @@ def update_output_stats_batch(
                             result[field_name][stat_config['output']].sum()
                         )
                         seq_set_size[field_name][stat] = (
-                            seq_set_size[field_name].get(stat, 0) +
-                            len(result[field_name][stat_config['output']])
+                                seq_set_size[field_name].get(stat, 0) +
+                                len(result[field_name][stat_config['output']])
```
Comment (ydudin3) [inline ludwig/models/model.py]: Move closing parenthesis to the line below.

File: ludwig/train.py [hunk 19]
```diff
@@ -119,15 +121,32 @@ def full_train(
            far are also resumed effectively cotinuing a previously interrupted
            training process.
     :type model_resume_path: filepath (str)
-    :param skip_save_progress_weights: Skips saving the weights at the end of
-           each epoch. If this is true, training cannot be resumed from the
-           exactly the state at the end of the previous epoch.
-    :type skip_save_progress_weights: Boolean
+    :param skip_save_model: Does not
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 20]
```diff
@@ -119,15 +121,32 @@ def full_train(
...
+           save model weights and hyperparameters each time the model
+           improves. By default Ludwig saves model weights after each epoch
+           the validation measure imrpvoes, but if the model is really big
+           that can be time consuming if you do not want to keep
+           the weights and just find out what performance can a model get
+           with a set of hyperparameters, use this parameter to skip it,
+           but the model will not be loadable later on.
+    :type skip_save_model: Boolean
+    :param skip_save_progress: Doesn't save
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 21]
```diff
@@ -119,15 +121,32 @@ def full_train(
...
+           the model is really big that can be time consuming and will uses
+           twice as much space, use this parameter to skip it, but training
+           cannot be resumed later on.
+    :type skip_save_progress: Boolean
     :param skip_save_processed_input: If a CSV dataset is provided it is
            preprocessed and then saved as an hdf5 and json to avoid running
            the preprocessing again. If this parameter is False,
            the hdf5 and json file are not saved.
     :type skip_save_processed_input: Boolean
+    :param skip_save_log: Does not save TensorBoard
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 22]
```diff
@@ -331,10 +355,27 @@ def train(
     :param model_load_path: If this is specified the loaded model will be used
            as initialization (useful for transfer learning).
     :type model_load_path: filepath (str)
-    :param skip_save_progress_weights: Skips saving the weights at the end of
-           each epoch. If this is true, training cannot be resumed from the
-           exactly the state at the end of the previous epoch.
-    :type skip_save_progress_weights: Boolean
+    :param skip_save_model: Does not
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 23]
```diff
@@ -331,10 +355,27 @@ def train(
...
+           save model weights and hyperparameters each time the model
+           improves. By default Ludwig saves model weights after each epoch
+           the validation measure imrpvoes, but if the model is really big
+           that can be time consuming if you do not want to keep
+           the weights and just find out what performance can a model get
+           with a set of hyperparameters, use this parameter to skip it,
+           but the model will not be loadable later on.
+    :type skip_save_model: Boolean
+    :param skip_save_progress: Doesn't save
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 24]
```diff
@@ -331,10 +355,27 @@ def train(
...
+           progress each epoch. By default Ludwig saves weights and stats
+           after each epoch for enabling resuming of training, but if
+           the model is really big that can be time consuming and will uses
+           twice as much space, use this parameter to skip it, but training
+           cannot be resumed later on.
+    :type skip_save_progress: Boolean
+    :param skip_save_log: Does not save TensorBoard
```
Comment (ydudin3) [inline ludwig/train.py]: Disables saving ...

File: ludwig/train.py [hunk 25]
```diff
@@ -591,13 +634,36 @@ def cli(sys_argv):
         help='path of a the model directory to resume training of'
     )
     parser.add_argument(
-        '-sspw',
-        '--skip_save_progress_weights',
+        '-ssm',
+        '--skip_save_model',
+        action='store_true',
+        default=False,
+        help='does not save weights each time the model imrpoves. '
```
Comment (ydudin3) [inline ludwig/train.py]: disables saving ...

File: ludwig/train.py [hunk 26]
```diff
@@ -591,13 +634,36 @@ def cli(sys_argv):
...
              'weights after each epoch for enabling resuming of training, but '
              'if the model is really big that can be time consuming and will '
              'save twice as much space, use this parameter to skip it.'
     )
+    parser.add_argument(
+        '-ssl',
+        '--skip_save_log',
+        action='store_true',
+        default=False,
+        help='does not save TensorBoard logs. By default Ludwig saves '
```
Comment (ydudin3) [inline ludwig/train.py]: disables saving ...

File: mkdocs/docs/api.md [hunk 27]
```diff
@@ -427,20 +429,33 @@ __Inputs__
...
    resume training of
-- __skip_save_progress_weights__ (bool, default: `False`): doesn't save
-   weights after each epoch. By default Ludwig saves weights after
-   each epoch for enabling resuming of training, but if the model is
-   really big that can be time consuming and will save twice as much
-   space, use this parameter to skip it.
-- __dataset_type__ (string, default: `'default'`): determines the type
-   of preprocessing will be applied to the data. Only `generic` is
-   available at the moment
+- __skip_save_model__ (bool, default: `False`): does not
```
Comment (ydudin3) [inline mkdocs/docs/api.md]: Same

File: mkdocs/docs/api.md [hunk 28]
```diff
@@ -427,20 +429,33 @@ __Inputs__
...
+   save model weights and hyperparameters each time the model
+   improves. By default Ludwig saves model weights after each epoch
+   the validation measure imrpvoes, but if the model is really big
+   that can be time consuming if you do not want to keep
+   the weights and just find out what performance can a model get
+   with a set of hyperparameters, use this parameter to skip it,
+   but the model will not be loadable later on.
+- __skip_save_progress__ (bool, default: `False`): doesn't save
```
Comment (ydudin3) [inline mkdocs/docs/api.md]: Same

File: mkdocs/docs/api.md [hunk 29]
```diff
@@ -427,20 +429,33 @@ __Inputs__
...
+   progress each epoch. By default Ludwig saves weights and stats
+   after each epoch for enabling resuming of training, but if
+   the model is really big that can be time consuming and will uses
+   twice as much space, use this parameter to skip it, but training
+   cannot be resumed later on.
+- __skip_save_log__ (bool, default: `False`): does not save TensorBoard
```
Comment (ydudin3) [inline mkdocs/docs/api.md]: Same

File: mkdocs/docs/user_guide.md [hunk 30]
```diff
@@ -90,12 +90,25 @@ optional arguments:
                         path of a pretrained model to load as initialization
   -mrp MODEL_RESUME_PATH, --model_resume_path MODEL_RESUME_PATH
                         path of a the model directory to resume training of
-  -sspw SKIP_SAVE_PROGRESS_WEIGHTS, --skip_save_progress_weights SKIP_SAVE_PROGRESS_WEIGHTS
-                        doesn't save weights after each epoch. By default
+  -ssm, --skip_save_model
+                        does not save weights each time the model imrpoves. By
```
Comment (ydudin3) [inline mkdocs/docs/user_guide.md]: Same

File: mkdocs/docs/user_guide.md [hunk 31]
```diff
@@ -90,12 +90,25 @@ optional arguments:
...
+                        default Ludwig saves weights after each epoch the
+                        validation measure imrpvoes, but if the model is
+                        really big that can be time consuming if you do not
+                        want to keep the weights and just find out what
+                        performance can a model get with a set of
+                        hyperparameters, use this parameter to skip it.
+  -ssp, --skip_save_progress
+                        does not save weights after each epoch. By default
```
Comment (ydudin3) [inline mkdocs/docs/user_guide.md]: Same

File: mkdocs/docs/user_guide.md [hunk 32]
```diff
@@ -90,12 +90,25 @@ optional arguments:
...
                         ludwig saves weights after each epoch for enabling
                         resuming of training, but if the model is really big
                         that can be time consuming and will save twice as much
                         space, use this parameter to skip it.
+  -ssl, --skip_save_log
+                        does not save TensorBoard logs. By default Ludwig
```
Comment (ydudin3) [inline mkdocs/docs/user_guide.md]: Same

File: ludwig/models/model.py (hunk 18 above)
Comment (w4nderlust) [reply ludwig/models/model.py]: that right parenthesis closes len(), so ti should be alright I believe

**Pull request merged**